packages/babel-types/src/validators/isReferenced.js

```diff
@@ -134,6 +134,18 @@ export default function isReferenced(
     // no: type X = { NODE: OtherType }
     case "ObjectTypeProperty":
       return parent.key !== node;
+
+    // no: enum X { NODE }
+    case "TSEnumMember":
+      return false;
+
+    // no: { NODE: NODE }
```

Comment:
This should be `{ NODE: value }`, since only the key isn't referenced.

Here, the type `string` _is_ a reference:

```js
type X = { foo: string }
```

Reply:
Updated, appreciate the hand-holding. 😄

Going to add a couple more tests to babel-types instead of relying on the tests in plugin-transform-typescript.